enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -110,6 +112,16 @@ const InteractiveDashboardInner = ({
   );
 };
 
-export const InteractiveDashboard = withPublicComponentWrapper(
-  InteractiveDashboardInner,
-);
+export const InteractiveDashboard =
+  withPublicComponentWrapper<InteractiveDashboardProps>(
+    ({ dashboardId, ...rest }) => {
+      const id = useValidIdForEntity({
```

Comment:
A bit curious about why we don't add `useValidIdForEntity` into the `InteractiveDashboardInner` component itself, if the input `id` would support both stable ids and numerical ids?

Reply:
Ah! I think there's some misunderstanding here (most likely caused by me, don't worry). The hook will take in anything - whether that's an entity ID or a numerical ID or whatever, and if the entity ID can't be found then we return the value that was passed in. So basically if we were to pass a numerical ID then the hook would return the numerical ID. 

I'll try to make this clearer in the code so it's more obvious